### PR TITLE
Update Shared Core Code From Pro Plugin

### PR DESCRIPTION
**Pull request by spencerfinnell**
See #64

### CODE REVIEW

File: includes/core/abstracts/form.php [hunk 1]
```diff
@@ -221,12 +258,12 @@ public function set_post_meta_settings() {
 
 		/* one-time payment options */
 
-		$this->amount = simpay_convert_amount_to_cents( simpay_get_filtered( 'amount', simpay_get_saved_meta( $this->id, '_amount', '100' ), $this->id ) );
```
Comment (spencerfinnell) [inline includes/core/abstracts/form.php]: This should be double checked.

File: includes/core/abstracts/form.php [hunk 2]
```diff
@@ -221,12 +258,12 @@ public function set_post_meta_settings() {
...
+		$this->amount = simpay_get_filtered( 'amount', simpay_get_saved_meta( $this->id, '_amount', simpay_global_minimum_amount() ), $this->id );
```
Comment (spencerfinnell) [inline includes/core/abstracts/form.php]: This should be double checked.

File: includes/core/admin/metaboxes/settings.php [hunk 3]
```diff
@@ -178,22 +205,17 @@ public static function save( $post_id, $post ) {
...
 
+		// TODO Rewrite. Hard to read.
+
 		// Amount
-		if ( $is_zero_decimal ) {
-			$amount = isset( $_POST['_amount'] ) ? sanitize_text_field( $_POST['_amount'] ) : ( false !== get_post_meta( $post_id, '_amount', true ) ? get_post_meta( $post_id, '_amount', true ) : '100' );
-		} else {
-			$amount = isset( $_POST['_amount'] ) ? sanitize_text_field( $_POST['_amount'] ) : ( false !== get_post_meta( $post_id, '_amount', true ) ? get_post_meta( $post_id, '_amount', true ) : '1' );
-		}
+		$amount = isset( $_POST['_amount'] ) ? sanitize_text_field( $_POST['_amount'] ) : ( false !== get_post_meta( $post_id, '_amount', true ) ? get_post_meta( $post_id, '_amount', true ) : simpay_global_minimum_amount() );
```
Comment (spencerfinnell) [inline includes/core/admin/metaboxes/settings.php]: This should be double checked.

File: includes/core/payments/details.php [hunk 4]
```diff
@@ -69,7 +71,7 @@ public function set_tags() {
 
 			'total-amount' => array(
 				'type'  => array( 'all' ),
-				'value' => isset( $this->payment->charge->amount ) ? simpay_formatted_amount( $this->payment->charge->amount, $this->payment->get_currency() ) : '',
+				'value' => isset( $this->payment->charge->amount ) ? simpay_format_currency( simpay_convert_amount_to_dollars( $this->payment->charge->amount ), $this->payment->get_currency() ) : '',
```
Comment (spencerfinnell) [inline includes/core/payments/details.php]: This should be double checked.

File: includes/core/forms/default-form.php [hunk 5]
```diff
@@ -101,19 +134,20 @@ public function print_custom_fields() {
...
-			foreach ( $this->custom_fields as $k => $v ) {
 
-				switch ( $v['type'] ) {
-					case 'payment_button':
-						$html .= Fields\Payment_Button::html( $v );
-						break;
-					case has_filter( 'simpay_custom_fields' ):
-						$html .= apply_filters( 'simpay_custom_fields', $html, $v );
-						break;
-				}
```
Comment (spencerfinnell) [inline includes/core/forms/default-form.php]: This broke the payment button output.

File: includes/core/abstracts/form.php (hunk 2 above)
Comment (pderksen) [reply includes/core/abstracts/form.php]: Looks correct. Part of amount conversion refactoring done in early 2018 in Pro.
https://github.com/wpsimplepay/WP-Simple-Pay-Pro-3/commit/fc7474f73217f19868c735b2b78d67e8568a98e2#diff-e560c0ff2a130a6267e5401b99a13c37

File: includes/core/admin/metaboxes/settings.php (hunk 3 above)
Comment (pderksen) [reply includes/core/admin/metaboxes/settings.php]: Looks correct. Part of amount conversion refactoring done in early 2018 in Pro.
https://github.com/wpsimplepay/WP-Simple-Pay-Pro-3/commit/fc7474f73217f19868c735b2b78d67e8568a98e2#diff-e560c0ff2a130a6267e5401b99a13c37

File: includes/core/payments/details.php (hunk 4 above)
Comment (pderksen) [reply includes/core/payments/details.php]: Looks correct. Part of amount conversion refactoring done in early 2018 in Pro.

https://github.com/wpsimplepay/WP-Simple-Pay-Pro-3/commit/73be1bc9aa1a60ef7401ddad4bdd43fcc2ebd4c7#diff-c456e49aee98642a1d8fac8d13b9b0ca
https://github.com/wpsimplepay/WP-Simple-Pay-Pro-3/commit/ec9c0fed173c1b078b96c6995d0416868e0fff95#diff-c456e49aee98642a1d8fac8d13b9b0ca

File: includes/core/forms/default-form.php (hunk 5 above)
Comment (pderksen) [reply includes/core/forms/default-form.php]: Unsure here.

See https://github.com/wpsimplepay/WP-Simple-Pay-Pro-3/commits/master/includes/core/forms/default-form.php

Also https://github.com/wpsimplepay/WP-Simple-Pay-Pro-3/issues/576

**Comment by spencerfinnell** [comment]
@pderksen Immediate glaring things out of the way and basic testing seems good to me. Up to you if you want to merge these changes in to `release/2.1.0` for testing or continue testing on this branch.

**Comment by pderksen** [comment]
@spencerfinnell Go ahead and merge into the release branch when you're comfortable with it and we can do all our testing from there.

**Comment by pderksen** [comment]
Merging and we can test on the release branch.

**Pull request merged**